src/quick-lint-js/parse.h

```diff
@@ -137,6 +142,24 @@ class parser {
     this->have_unimplemented_token_jmp_buf_ = false;
     return ok;
   }
+  // Returns true if parsing succeeded without QLJS_PARSER_DEPTH_LIMIT_EXCEEDED
+  // being called.
+  //
+  // Returns false if QLJS_PARSER_DEPTH_LIMIT_EXCEEDED was called.
+  template <QLJS_PARSE_VISITOR Visitor>
+  bool parse_and_visit_module_catching_parser_depth_limit_exceeded(Visitor &v) {
```

Comment:
Let's not make two functions (`parse_and_visit_module_catching_unimplemented` and `parse_and_visit_module_catching_parser_depth_limit_exceeded`). Let's join the two. They do basically the same thing anyway.

Call the function `parse_and_visit_module_catching_fatal_parse_errors` and the variable `fatal_parse_error_jmp_buf_` or something.